src/cargo/sources/registry/local.rs

```diff
@@ -129,8 +129,10 @@ impl<'cfg> RegistryData for LocalRegistry<'cfg> {
         // We don't actually need to download anything per-se, we just need to
         // verify the checksum matches the .crate file itself.
         let actual = Sha256::new().update_file(&crate_file)?.finish_hex();
-        if actual != checksum {
-            anyhow::bail!("failed to verify the checksum of `{}`", pkg)
+        if let Some(checksum) = checksum {
+            if actual != checksum {
+                anyhow::bail!("failed to verify the checksum of `{}`", pkg)
+            }
```

Comment:
I am thinking if we could have this simple change, so that we don't need to make too many changes on other structs and interfaces. As a bonus, we can also keep the error `no hash listed for {pkg}` in a sinple place. Thoughts?



```suggestion
        // If a checksum of a package from local registry is empty,
        // we assume it comes from a git dependency and can be skipped.
        // (Probably be generated by tool `cargo-local-registry`)
        if !checksum.is_empty() && actual != checksum {
            anyhow::bail!("failed to verify the checksum of `{}`", pkg)
```